### ISSUE TITLE
Infinite loop occurs when parentheses are included in a comment

### ISSUE THREAD

**Issue by mugisoba** [xoofx/CppAst#17]
I found an infinite loop on v0.7.0.

```cpp
// [infinite loop)
void x() {}
```

If comments contain parentheses,
this loop cannot be escaped because `parenCount` has never become zero.

```cs
// CppModelBuilder.cs Line 2133
var parenCount = 1;
for (var lastBegin = begin; parenCount != 0; lastBegin = begin)
{
    if (TokenIsBefore(begin, "("))
        --parenCount;
    else if (TokenIsBefore(begin, ")"))
        ++parenCount;

    begin = GetPrevLocation(begin, 1);  // GetPrevLocation() returns CXSourceLocation.Null
}
```

**Comment by xoofx** [comment]
@wackoisgod was worried about a few parts in the PR about that, should have raised my concerns! 😅
Could you have a look?

**Comment by wackoisgod** [comment]
I shall take a look!

**Comment by wackoisgod** [comment]
Should be resolved in PR #18

**Comment by mugisoba** [comment]
It works perfectly 🎉 
Thank you very much 👍
